code/game/turfs/simulated/wall_icon.dm

```diff
@@ -105,14 +109,17 @@
 					if(success)
 						wall_dirs += get_dir(src, T)
 						if(get_dir(src, T) in global.cardinal)
+							for(var/nb_type in global.wall_fullblend_objects)
+								if(istype(O, nb_type))
+									break
```

Comment:
This will only break out of the highest `for`, as written the loop does nothing but burn cycles.